staging/src/k8s.io/client-go/rest/request.go

```diff
@@ -151,6 +151,7 @@ func NewRequest(c *RESTClient) *Request {
 		r.SetHeader("Accept", c.content.AcceptContentTypes)
 	case len(c.content.ContentType) > 0:
 		r.SetHeader("Accept", c.content.ContentType+", */*")
+		r.SetHeader("Content-Type", r.c.content.ContentType)
```

Comment:
setting Content-Type when we don't know if we're even sending a body in the request is not correct